price_feed_poller.py

```diff
@@ -172,19 +222,41 @@ def main(args):
     gas = config["network"].get("gas", 4000000)
     # Get gas price, defaults to 100 gwei:
     gas_price = config["network"].get("gas_price", 100000000000) 
+    # Get HTTP-JSON-RPC waiting timeout (in secs):
+    tx_waiting_timeout_secs = config["network"].get("tx_waiting_timeout_secs", 120) 
+    # Get HTTP-JSON-RPC polling latency timeout (in secs):
+    tx_polling_latency_secs = config["network"].get("tx_polling_latency_secs", 5) 
     # Load the WRB contract:
     wrbcontract = wrb(w3, config)
-    current_block = w3.eth.blockNumber
+
+    # Try connecting to JSON-RPC provider and get latest block:
+    try:
+      current_block = w3.eth.blockNumber
+      print(f"Connected to {provider}")
```

Comment:
Can't see how this statement can fail.

Reply:
`w3.eth.blockNumber` can actually fail. It's not a matter if a call makes changes on Ethereum storage. It's a matter that after sending `eth_blockNumber`, the HTTP response could not arrive before timeout.

Reply:
I'm assuming that this statment does nothing in runtime but copying the value of `w3.eth.blockNumber` to `current_block`, so what's the possible exception here?

Reply:
Apparently, Python's web3 library performs lazy binding with the remote provider, only trying the connection on first attempt to read any data from the remote network. 

Proof for this:

- If you move `current_block = w3.eth.blockNumber` out from the try/except block, and run the poller without a running instance of gateway you get a long and ugly unhandled exception message: "[..........] requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=3000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000019D6B2875B0>: Failed to establish a new connection: [WinError 10061] No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión'))"

- Levaing `current_block = w3.eth.blockNumber`, the exception is captured and you get this other message: "Fatal: connection failed to http://127.0.0.1:3000"